src/otx/core/metrics/ap_3d.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTX  metric used for 3D object detection tasks."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from torch import Tensor
+from torchmetrics import Metric
+
+from otx.core.metrics.kitti_3d_eval import get_coco_eval_result
+
+if TYPE_CHECKING:
+    from otx.core.types.label import LabelInfo
+
+
+class KittiMetric(Metric):
```

Comment:
We don't have other datasets right now. Actually, the number of data formats and datasets in that field  are very limited. It can be used only for kitti format.

Reply:
In other repositories such as OpenMMLab3D, they also use the similar metric name and same implementation